std/algorithm/mutation.d

```diff
@@ -2389,6 +2374,49 @@ void swap(T)(ref T lhs, ref T rhs) if (is(typeof(lhs.proxySwap(rhs))))
     lhs.proxySwap(rhs);
 }
 
+/**
+Swaps two elements in-place of a random access range `r`,
+specified by their indices `i1` and `i2`.
+
+Params:
+    r  = a $(XREF_PACK_NAMED range,primitives,isRandomAccess,random access range)
+        with swappable elements
+    i1 = first index
+    i2 = second index
+*/
+void swapAt(R)(R r, size_t i1, size_t i2)
```

Comment:
@DmitryOlshansky see `moveAt` for example: https://github.com/D-Programming-Language/phobos/blob/75702dc9957996d4cdb87c65b3104ac8d0ea5e67/std/range/primitives.d#L1946

It has not any constaints